### PR TITLE
Add tree component 

### LINKED ISSUE #1673: Add Antd tree component
[Neeto-testify](https://github.com/bigbinary/neeto-testify-web) currently has a custom implementation of the tree structure to display the[ suites and their test cases](https://spinkart.neetotestify.net/projects/37167065-8ea9-461c-9ead-9b043065814f/test-cases). `Antd` has a [tree component](https://ant.design/components/tree) that has many built-in features. Is it possible to include  `Antd` `tree` in `neeto-ui`?

image

### PR DESCRIPTION
**Pull request by AkhilXavier95**
Fixes #1673

#### Description
- Added: `Tree` component using ant-d.

#### Checklist
- [x] I have made corresponding changes to the documentation.
- [x] I have updated the types definition of modified exports.
- [ ] I have verified the functionality in some of the neeto web-apps.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary label (patch/minor/major - If package publish
      is required).

### CODE REVIEW

File: src/components/Tree/index.jsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    defaultSelectedKeys={defaultSelectedKeys}
+    draggable={draggable}
+    showLine={showLine}
+    switcherIcon={SwitcherIcon}
+    treeData={treeData}
+    onCheck={onCheck}
+    onSelect={onSelect}
+    {...otherProps}
+  />
+);
```
Comment (ajmaln) [inline src/components/Tree/index.jsx:36]:
```suggestion
const Tree = (props) => (
  <TreeComponent
    switcherIcon={SwitcherIcon}
    {...props}
    className={classnames("neeto-ui-tree draggable-tree", props.className)}
  />
);
```

File: src/components/Tree/index.jsx
Comment (ajmaln) [inline src/components/Tree/index.jsx:1]: Let's add a type for the component

**Review by ajmaln — APPROVED** [review]
@AkhilXavier95 _a Added few suggestions, please assign to @praveen-murali-ind for reviewing styles after updating

**Comment by AkhilXavier95** [comment]
@ajmaln _a added the types and updated the props for the tree component.

**Comment by ajmaln** [comment]
@praveen-murali-ind _a Please review the styles

**Comment by praveen-murali-ind** [comment]
@goutham-subramanyam _a Could you please update the styling to ensure we are following the new CSS-IN-JS approach? Currently dark mode is not working properly for this component.

**Comment by goutham-subramanyam** [comment]
@praveen-murali-ind _a I have made the changes. Please review.

**Review by praveen-murali-ind — APPROVED** [review]
LGTM

**Comment by praveen-murali-ind** [comment]
@AkhilXavier95 _A Please resolve the conflicts.

**Pull request merged**